File: AddressBook/react-client/src/dtos/Label.ts
import Common from "../utility/Common";
import { PagedReq } from "./PagedReq";

export interface LabelRes {
  labelId?: string;
  name?: string;
}

export class LabelReqEdit {
  name?: string = "";
}

export class LabelReqSearch extends PagedReq {
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy,
      searchText = "",
    }: PagedReq,
    {}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
  }
}

File: AddressBook/Dtos/Dtos/Label.cs
using AddressBook.Common.Paging;
using System.ComponentModel.DataAnnotations;

namespace AddressBook.Dtos
{
    public class LabelRes
    {
        public int LabelId { get; set; }
        public string? Name { get; set; }
    }

    public class LabelReqEdit
    {
        [Required, MaxLength(20)]
        public string? Name { get; set; }
    }

    public class LabelReqSearch : PagedReq
    {

    }
}
